### ISSUE TITLE
Declaration errors that block emit are not reported

### ISSUE THREAD

**Issue by vladima** [Microsoft/TypeScript#6198]
```ts
// t.ts
type T = { x : number }
export interface I {
	f: T;	
}
```
Compiling this file with these flags `-d --noEmitOnError -m commonjs` yields no output file and no errors are reported
